### PR TITLE
Extension telemetry (include versions)

### PR DESCRIPTION
**Pull request by derekbekoe**

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5512
This is an **experimental preview** for @microsoft users.

File: src/azure-cli-core/azure/cli/core/telemetry.py [hunk 1]
```diff
@@ -146,8 +146,9 @@ def _get_azure_cli_properties(self):
         self.set_custom_properties(result, 'PythonVersion', platform.python_version())
         self.set_custom_properties(result, 'ModuleCorrelation', self.module_correlation)
         self.set_custom_properties(result, 'ExtensionName', self.extension_name)
+        self.set_custom_properties(result, 'ExtensionVersion', self.extension_version)
```
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/telemetry.py]: Can't we just combine the name and version in one string?

File: src/azure-cli-core/azure/cli/core/telemetry.py [hunk 2]
```diff
@@ -176,6 +177,11 @@ def module_version(self):
     def product_version(self):
         return _get_core_version()
 
+    @property
+    def extension_version(self):
+        from azure.cli.core.extension import get_extension
+        return get_extension(self.extension_name).version if self.extension_name else None
```
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/telemetry.py]: Move this logic to the place where `set_command_details` is called. The `telemetry` module should not have knowledge (dependencies) of extension's logic.

**Comment by derekbekoe** [comment]
@troydai feedback addressed. Please take another look.

**Pull request merged**